compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3741,6 +3747,10 @@ object Types extends TypeUtils {
       case that: OrType => tp1.eq(that.tp1) && tp2.eq(that.tp2) && isSoft == that.isSoft
       case _ => false
     }
+
+    override protected def iso(that: Any, bs: BinderPairs) = that match
+      case that: OrType => tp1.equals(that.tp1, bs) && tp2.equals(that.tp2, bs)
```

Comment:
Not sure if it should also require
```suggestion
      case that: OrType => tp1.equals(that.tp1, bs) && tp2.equals(that.tp2, bs) && isSoft == that.isSoft
```
as in `eql` ?

Reply:
Well it's going to be the more conservative approach to include it.